src/java.base/unix/classes/jdk/internal/loader/ClassLoaderHelper.java

```diff
@@ -36,10 +42,15 @@ private ClassLoaderHelper() {}
      * Returns an alternate path name for the given file
      * such that if the original pathname did not exist, then the
      * file may be located at the alternate location.
-     * For most platforms, this behavior is not supported and returns null.
+     * For AIX, this replaces the final .so suffix with .a
      */
     static File mapAlternativeName(File lib) {
-        return null;
+        String name = lib.toString();
```

Comment:
This code doesn't seem specific to AIX, but will affect all platforms.

Reply:
True, this is going to affect unix-alike platforms except OSX and Windows which have their own version of `ClassLoaderHelper`. It is not expected this method works with other unix-alike platforms except `AIX` since currently it returns `null` anyway.
It is possible to create a dedicated `ClassLoaderHelper.java` for `AIX` at `java.base/aix/classes/jdk/internal/loader/`, my thought was to reuse this general version until it is required by other unix-alike systems.

Reply:
It needs to be AIX specific, we don't want other platforms trying to load .a files.
Also, I'm thinking mapLibaryName() should continue to return .a for compatibility with Java 8, and this method should map to .so

Reply:
Not against the idea to load `.a` files first for back compatibility, however Hotspot JDK8+ return names with `.so` suffix at `mapLibaryName()`, unless OpenJDK community agree to change it to `.a`,  this method should map to the other suffix instead.
OpenJ9 `mapAlternativeName()` need match Hotspot to take advantage of this JDK15+ JCL patch. 
https://github.com/eclipse/openj9/pull/9489 has restored OpenJ9 JDK8 to previous `.a` suffix.
OpenJ9 JDK11 currently returns `.so` suffix.
There were some discussion about the choice of `.a` or `.so` at https://github.com/eclipse/openj9-docs/issues/561.
There is an internal request to get clarification about `AIX hotspot mapLibaryName()`, I think OpenJ9 should match RI behaviours and contribute this PR to OpenJDK.

Reply:
Created `java.base/aix/classes/jdk/internal/loader/ClassLoaderHelper.java` which is based on `OSX` version and returns `.a` suffix for now.

Reply:
I'm wondering if the IBM Java version is the more popular version for AIX over OpenJDK, and OpenJ9 should keep compatibility with it for JDK 11+.

Reply:
If `mapLibaryName()` returns `.a` suffix and `mapAlternativeName()` handles `.so` in this PR, it probably won't be accepted by OpenJDK unless its `mapLibaryName()` is changed to match `OpenJ9` behaviours.